rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
 		if (selection.size > 1) {
 			return selection.toList.map[testCaseListFromSelection].flatten.toList
 		} else {
-			if (selection.firstElement instanceof IFolder) {
-				return selection.firstElement.testCaseListFromSelection.toList
-			}
-			return #[launchShortcutUtil.getQualifiedNameForTestInTcl(selection.firstElement as IResource).toString]
+			return selection.firstElement.testCaseListFromSelection.toList
 		}
 	}
 
 	def Iterable<String> getTestCaseListFromSelection(Object sel) {
```

Comment:
From the name of the method I would expect that it operates on an `IStructuredSelection`. Also the parameter name suggests that. Can we pass in the whole selection here and let the method deal with it instead of the caller?